File: conformance/tests/arithmetic/arithmetic.runar.zig
const runar = @import("runar");

pub const Arithmetic = struct {
    pub const Contract = runar.SmartContract;

    target: i64,

    pub fn init(target: i64) Arithmetic {
        return .{ .target = target };
    }

    pub fn verify(self: *const Arithmetic, a: i64, b: i64) void {
        const sum = a + b;
        const diff = a - b;
        const prod = a * b;
        const quot = @divTrunc(a, b);
        const result = sum + diff + prod + quot;
        runar.assert(result == self.target);
    }
};

File: conformance/tests/arithmetic/arithmetic.runar.go
//go:build ignore

package contract

import "runar"

type Arithmetic struct {
	runar.SmartContract
	Target runar.Int `runar:"readonly"`
}

func (c *Arithmetic) Verify(a runar.Int, b runar.Int) {
	sum := a + b
	diff := a - b
	prod := a * b
	quot := a / b
	result := sum + diff + prod + quot
	runar.Assert(result == c.Target)
}
